### PR TITLE
CLI refactor

### PR DESCRIPTION
**Pull request by Fernerkundung**
Refactoring of the CLI as discussed in #115 

- merged `sentinel search` and `sentinel download` into `sentinelsat`
- replaced positional arguments with `-u, --user`, `-p, --password` (required) and `-g, --geometry`
- `--geometry` still takes the path to a GeoJSON file
- `--uuid` keyword, takes a list of IDs - replacement for `sentinel download`
-   `--name` to search/download by product name - supports wildcards

### CODE REVIEW

**Comment by Fernerkundung** [comment]
This PR is ready to merge.

@valgur would you mind giving this a quick review, since you opened #115?

File: sentinelsat/scripts/cli.py [hunk 1]
```diff
@@ -115,21 +114,26 @@ def search(
...
-    products = api.query(wkt, start, end, order_by=order_by, limit=limit, **search_kwargs)
+    if geometry is not None:
+        wkt = geojson_to_wkt(read_geojson(geometry))
+
+    if uuid is not None:
+        uuid_list = [x.strip() for x in uuid.split(',')]
+        products = {}
+        for productid in uuid_list:
+            try:
+                products[productid] = api.get_product_odata(productid)
```
Comment (valgur) [inline sentinelsat/scripts/cli.py]: `get_product_odata` will produce different set of attributes than `query`, which is a slight inconsistency when used with `--footprints`. This is a pretty minor use case, though, which we can safely ignore, though, I think.

File: sentinelsat/scripts/cli.py [hunk 2]
```diff
@@ -115,21 +114,26 @@ def search(
...
+            except SentinelAPIError as e:
+                if 'Invalid key' in e.msg:
+                    logger.error('No product with ID \'%s\' exists on server', productid)
+    elif name is not None:
+        search_kwargs["identifier"] = name
```
Comment (valgur) [inline sentinelsat/scripts/cli.py]: Consider using `_query_names()` here. This would also enable a list of names to be used as input, similar to `--uuid`.

Comment (valgur) [inline sentinelsat/scripts/cli.py]: `_query_names()` does somewhat more than is necessary, though. I would simplify it by moving the 
```python
        # Group the products
        output = OrderedDict((name, dict()) for name in names)
        for id, metadata in products.items():
            name = metadata['identifier']
            output[name][id] = metadata
```
part to `check_files()`. I can take care of it myself, if you wish.

**Pull request merged**